odiglet/pkg/ebpf/manager.go

```diff
@@ -0,0 +1,371 @@
+		m.Logger.Error(err, "error deleting instrumentation instance", "pod", details.Pod.Name, "pid", pid)
+	}
+
+	delete(m.DetailsByPid, pid)
+}
+
+func (m *Manager) handleProcessExecEvent(ctx context.Context, e detector.ProcessEvent) error {
+	if _, found := m.DetailsByPid[e.PID]; found {
+		// this can happen if we have multiple exec events for the same pid (chain loading)
+		// TODO: better handle this?
+		// this can be done by first closing the existing instrumentation,
+		// and then creating a new one
+		m.Logger.Info("instrumentation already exists for pid", "pid", e.PID)
+		return nil
+	}
+
+	// get the corresponding pod object for this process event
+	pod, err := m.podFromProcEvent(e)
+	if err != nil {
+		return fmt.Errorf("failed to get pod from process event: %w", err)
+	}
+
+	containerName, found := containerNameFromProcEvent(e)
+	if !found {
+		return ErrContainerNameNotReported
+	}
+
+	// get the language and sdk for this process event
+	// based on the pod spec and the container name from the process event
+	// TODO: We should have all the required information in the process event
+	// to determine the language - hence in the future we can improve this
+	lang, sdk, err := m.languageSdk(pod, containerName)
+	if err != nil {
+		return fmt.Errorf("failed to get language and sdk: %w", err)
+	}
+
+	factory, found := m.Factories[FactoryID{Language: lang, OtelSdk: sdk}]
+	if !found {
+		return ErrNoInstrumentationFactory
+	}
+
+	workload, err := workloadUtils.PodWorkloadObject(ctx, pod)
+	if err != nil {
+		return fmt.Errorf("failed to get workload object: %w", err)
+	}
+
+	// Fetch initial config based on the InstrumentationConfig CR
+	sdkConfig := m.instrumentationSDKConfig(ctx, workload, lang, types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name})
+	// we should always have config for this event.
+	// if missing, it means that either:
+	// - the config will be generated later due to reconciliation timing in instrumentor
+	// - just go deleted and the pod (and the process) will go down soon
+	// TODO: sync reconcilers so ins config is guaranteed be created before the webhook is enabled
+	//
+	// if sdkConfig == nil {
+	// 	m.Logger.Info("no sdk config found for language", "language", lang, "pod", pod.Name)
+	// 	return nil
+	// }
+
+	settings := Settings{
+		// TODO: respect reported name annotation (if present) - to override the service name
+		// refactor from opAmp code
+		ServiceName:        workload.Name,
+		// TODO: add container name
+		ResourceAttributes: utils.GetResourceAttributes(workload, pod.Name),
+		InitialConfig:      sdkConfig,
+	}
+
+	inst, err := factory.CreateInstrumentation(ctx, e.PID, settings)
+	if err != nil {
+		m.Logger.Error(err, "failed to initialize instrumentation", "language", lang, "sdk", sdk)
+
+		// write instrumentation instance CR with error status
+		err = m.updateInstrumentationInstanceStatus(ctx, pod, containerName, workload, e.PID, InstrumentationUnhealthy, FailedToInitialize, err.Error())
```

Comment:
Should API server writes happen on a different goroutine in order to not block the manager in case of a slow API server? may be overkill

Reply:
Discussed, and agreed to keep it in the same goroutine to reduce any possible race conditions.